core/vm/analysis.go

```diff
@@ -19,6 +19,45 @@ package vm
 import (
 )
 
+// fill in segment of operation information array for a block
+func analyzeBlock(ctx *callCtx, pc uint64) (*BlockInfo, error) {
+	info := NewBlockInfo(ctx.contract, pc)
+	code := ctx.contract.Code
+	jumpTable := ctx.interpreter.jt
+	for ; pc < uint64(len(code)); pc++ {
+		op := OpCode(code[pc])
+		operation := jumpTable[op]
+		if operation == nil {
+			return nil, &ErrInvalidOpCode{opcode: op}
+		}
+		info.minStack += operation.minStack
+		info.maxStack += operation.maxStack
```

Comment:
This won't work. You need a high-water mark and a low-water mark, so you know what the max 'stack pop' and 'stack push' value of the block is.

Reply:
Exactly. You need to track the relative stack height within the block and note the min and max value.

Reply:
Thanks, makes sense now.  If one of you knows how to fix easily let me know, else I'll get back to studying evmone.

Reply:
You can add together `operation.numPush` and `operation.numPop`, and then, outside of the loop, you calculate `minStack` and `maxStack` in the same way it is done for individual opcodes: `minStack` and `maxStack` there are derived from `numPush`, `numPop` and maximum stack depth

Reply:
Yeah, I was about to come hear and type, "What was I thinking?"

Reply:
There's another way, I think, 
but it will wait.